tests/unit/utils/number.test.ts

```diff
@@ -0,0 +1,22 @@
+import { describe, expect, it } from "vitest";
+
+import { padWithLeadingZeros } from "../../../src/utils/number.js";
+
+describe("padWithLeadingZeros", () => {
+  it("should pad the number with leading zeros when numberOfZeros is greater than the length of the number", () => {
+    expect(padWithLeadingZeros(5, 3)).toBe("005");
+    expect(padWithLeadingZeros(42, 5)).toBe("00042");
+    expect(padWithLeadingZeros(0, 4)).toBe("0000");
```

Comment:
Prefer parametrized tests over one test with multiple expects each for a different case.